CMake install ignores CMAKE_INSTALL_PREFIX for python components

Consider:
```
% mkdir build; cd build
% cmake .. -DCMAKE_INSTALL_PREFIX=/tmp/foo -DARB_WITH_PYTHON=ON
% make install
```

Current behaviour:
```
[…]
-- Installing: /usr/lib/python3.7/site-packages/arbor
CMake Error at python/cmake_install.cmake:54 (file):
  file INSTALL cannot make directory
  "/usr/lib/python3.7/site-packages/arbor": No such file or directory
Call Stack (most recent call first):
  cmake_install.cmake:92 (include)


make: *** [Makefile:74: install] Error 1
```

Note that the attempted path is not under `/tmp/foo`.

Expected behaviour:
```
[…]
-- Installing: /tmp/foo/lib/python3.7/site-packages/arbor
-- Installing: /tmp/foo/lib/python3.7/site-packages/arbor/VERSION
-- Installing: /tmp/foo/lib/python3.7/site-packages/arbor/__init__.py
-- Installing: /tmp/foo/lib/python3.7/site-packages/arbor/_arbor.cpython-37m-x86_64-linux-gnu.so
```

When installing in `/usr/local`, or in a stow directory, or in a staging install directory, or in fact in pretty much every circumstance where one wants to set `CMAKE_INSTALL_PREFIX`, all the installed files should be installed under that prefix unless there is a specific direction not to (e.g. via `ARB_PYTHON_LIB_PATH`).

--- a/python/CMakeLists.txt
+++ b/python/CMakeLists.txt
@@ -81,7 +81,15 @@ file(COPY "${PROJECT_SOURCE_DIR}/VERSION" DESTINATION "${python_mod_path}")
 # Set the installation path
 
 # Ask Python where it keeps its system (platform) packages.
-execute_process (COMMAND ${PYTHON_EXECUTABLE} -c "import sysconfig; print(sysconfig.get_path('platlib'))" OUTPUT_VARIABLE ARB_PYTHON_LIB_PATH_DEFAULT OUTPUT_STRIP_TRAILING_WHITESPACE)
+
+file(WRITE "${CMAKE_BINARY_DIR}/install-prefix" "${CMAKE_INSTALL_PREFIX}")
+execute_process(
+    COMMAND ${PYTHON_EXECUTABLE} -c
+    "import sys,sysconfig;pfx=sys.stdin.read();print(sysconfig.get_path('platlib',vars={} if pfx=='' else {'base':pfx,'platbase':pfx}))"
+    INPUT_FILE "${CMAKE_BINARY_DIR}/install-prefix"
+    OUTPUT_VARIABLE ARB_PYTHON_LIB_PATH_DEFAULT
+    OUTPUT_STRIP_TRAILING_WHITESPACE)
+
 # Default to installing in that path, override with user specified ARB_PYTHON_LIB_PATH
 set(ARB_PYTHON_LIB_PATH ${ARB_PYTHON_LIB_PATH_DEFAULT} CACHE PATH "path for installing Python module for Arbor.")
 message(STATUS "Python module installation path: ${ARB_PYTHON_LIB_PATH}")

--- a/python/CMakeLists.txt
+++ b/python/CMakeLists.txt
@@ -82,7 +82,12 @@ file(COPY "${PROJECT_SOURCE_DIR}/VERSION" DESTINATION "${python_mod_path}")
 
 # Ask Python where it keeps its system (platform) packages.
 
-file(WRITE "${CMAKE_BINARY_DIR}/install-prefix" "${CMAKE_INSTALL_PREFIX}")
+if(CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
+    file(WRITE "${CMAKE_BINARY_DIR}/install-prefix" "")
+else()
+    file(WRITE "${CMAKE_BINARY_DIR}/install-prefix" "${CMAKE_INSTALL_PREFIX}")
+endif()
+
 execute_process(
     COMMAND ${PYTHON_EXECUTABLE} -c
     "import sys,sysconfig;pfx=sys.stdin.read();print(sysconfig.get_path('platlib',vars={} if pfx=='' else {'base':pfx,'platbase':pfx}))"

